src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -780,12 +780,35 @@ public SolutionCompilationState WithDocumentAttributes<TArg>(
     }
 
     internal SolutionCompilationState WithDocumentTexts(ImmutableArray<(DocumentId documentId, SourceText text)> texts, PreservationMode mode)
```

Comment:
at this poitn, i'd like us to be doc'ing whihc methods allow for docIds that can be normal docIds, or sg doc ids.